hazelcast/src/main/java/com/hazelcast/internal/serialization/ObjectDataOutputStream.java

```diff
@@ -205,11 +214,27 @@ public void writeShortArray(short[] shorts) throws IOException {
     }
 
     @Override
+    public void writeUTFArray(String[] strings) throws IOException {
+        int len = strings != null ? strings.length : -1;
+        writeInt(len);
+        if (len > 0) {
+            for (String s : strings) {
+                writeUTF(s);
+            }
+        }
+    }
+
+    @Override
     public void writeUTF(String str) throws IOException {
-        if (utfBuffer == null) {
-            utfBuffer = new byte[UTF_BUFFER_SIZE];
+        int len = (str == null) ? 0 : str.length();
+        writeInt(len);
+        if (len > 0) {
+            for (int i = 0; i < len; i++) {
+                byte[] buffer = new byte[3];
```

Comment:
Optimization possibility: Could this "byte[] buffer = new byte[3];" be put out of the for loop and this buffer is reused?